src/core_plugins/kibana/public/management/sections/indices/create_index_pattern_wizard/components/empty_state/empty_state.js

```diff
@@ -40,27 +42,47 @@ export const EmptyState = ({
       <EuiFlexItem grow={false}>
         <EuiTitle>
           <EuiTextColor color="subdued">
-            <h2 style={{ textAlign: 'center' }}>Couldn&apos;t find any Elasticsearch data</h2>
+            <h2 style={{ textAlign: 'center' }}>
+              <FormattedMessage
+                id="kbn.management.createIndexPattern.emptyStateHeader"
+                defaultMessage="Couldn't find any Elasticsearch data"
+              />
+            </h2>
           </EuiTextColor>
         </EuiTitle>
         <EuiSpacer size="s"/>
         <EuiText>
           <p>
-            <EuiTextColor color="subdued">
-              You&apos;ll need to index some data into Elasticsearch before you can create an index pattern.
-            </EuiTextColor>
-            &nbsp;
-            <EuiLink
-              href="#/home/tutorial_directory"
-            >
-              Learn how
-            </EuiLink>
-            {' or '}
-            <EuiLink
-              href="#/home/tutorial_directory/sampleData"
-            >
-              get started with some sample data sets.
-            </EuiLink>
+            <FormattedMessage
+              id="kbn.management.createIndexPattern.emptyStateLabel.emptyStateDetail"
+              defaultMessage="{needToIndex} {learnHowLink} or {getStartedLink}"
+              values={{
+                needToIndex: (
```

Comment:
I think I would prefer the values object here to be declared as a constant above and then referenced here.  This would make things easier to read, and would also prevent a new object being created with each rerender.  See this for an explanation of why that might be bad: https://medium.com/@esamatti/react-js-pure-render-performance-anti-pattern-fb88c101332f

Reply:
@bmcconaghy Basically, this is common pattern for avoiding creation of new object, but I would not say, that in this case, because:
1) Moving such objects to constants will decrease readability of code in general
2) Some of values are React components too and use props. So if we move values to constant outside component we will not get props, if we will create it inside render it will be the same one
3) This case is envisaged by react-intl guys. And for avoiding rerendering they espacially use shallow compare for `values` props. (Here)[https://github.com/yahoo/react-intl/blob/master/src/components/message.js#L43]

Reply:
Fair enough on the performance point.  I actually find the FormattedMessage nested in FormattedMessage harder to read personally, as I get lost in the multiple levels of nesting, so would still like to see the values defined outside of the FormattedMessage and then referenced there.

Reply:
It feels like _the right_ solution for this readability problem would be integrating i18n into EUI components at some point, so that this:
```jsx
<EuiTextColor color="subdued">
  <FormattedMessage
    id="kbn.management.createIndexPattern.emptyStateLabel.needToIndexLabel"
    defaultMessage="You'll need to index ....."
  />
</EuiTextColor>
```
can be changed to:
```jsx
<EuiTextColor color="subdued" i18n-id="kbn.management.createIndexPattern.emptyStateLabel.needToIndexLabel">
	You'll need to index .....
</EuiTextColor>
```
But that's a bigger effort that spans several teams and i18n should prove itself in practice first.